core/modules/installer/installer.pages.inc

```diff
@@ -650,7 +650,18 @@ function installer_browser_installation_enable_form($form, &$form_state, $projec
   foreach ($projects as $project) {
     if ($project['type'] == 'module' && isset($modules[$project['name']])) {
       foreach ($modules as $module) {
-        if (isset($module->info['project']) && isset($modules[$project['name']]->info['project']) && $modules[$project['name']]->info['project'] == $module->info['project'] && $modules[$project['name']]->name != $module->name && !$module->info['hidden']) {
+        // Ignore stand-alone and hidden modules.
+        if (!isset($module->info['project']) || !isset($modules[$project['name']]->info['project']) || !empty($module->info['hidden'])) {
+          continue;
+        }
+        // The type key is required, skip over the violating module, but log.
```

Comment:
So here we're logging a warning when modules don't have a `type` set in their info file. But this function is very specific to the project installer. If we are to do this, I wonder if it should be done in a more general function. Something like [`_system_rebuild_module_data()`](https://api.backdropcms.org/api/backdrop/core%21modules%21system%21system.module/function/_system_rebuild_module_data/1) for example...?

Reply:
Hm, as far as I can tell, this is the only place where this message is needed. What's the benefit of a function that provides something that's only needed once?

Reply:
Well all the message does is inform the user that there are module(s) with no `type` value. Since this only happens if the user uses the project installer, I figured it might be more beneficial if this message were also logged for users who don't use the project installer - i.e. anytime Backdrop encounters a module without a `type`, log it. The more users who have access to these messages the more likely they'll get fixed, and therefore not be skipped in this function.

Reply:
I see... But doesn't that (enforce "type" for all modules) go far beyond the purpose of this Issue and PR? It really only tries to catch the PHP Notices on the project installer.

Reply:
> But doesn't that (enforce "type" for all modules) go far beyond the purpose of this Issue and PR?

If that's the case, then modules with no `type` should just be ignored by this PR and the logging code moved to a separate issue/PR.

Reply:
> If that's the case, then modules with no type should just be ignored by this PR and the logging code moved to a separate issue/PR.

Are you sure? We can't enforce things anyway, there will probably always be modules that do it wrong. Why should we "punish" the admin that installs such a module?
The message was something that has been requested. I've no problem with removing it, though, if that allows progress here.

Or would you prefer to still throw the notice until we fix the "type" problem in a stricter way?

I have to admit, I don't fully get what you're actually after. :wink: